day23/stringsRearrangement.py

```diff
@@ -0,0 +1,37 @@
+from itertools import permutations
+
+def stringsRearrangement(inputArray):
+    for perm in permutations(inputArray):
+        if neighborsDifferByOne(perm):
+            return True 
+    return False
+
+def neighborsDifferByOne(inputArray):
+    if len(inputArray) <= 1:
+        return True
+    return differByOne(inputArray[0], inputArray[1]) and neighborsDifferByOne(inputArray[2:])
+
+def differByOne(word, anotherWord):
```

Comment:
if you wanted to be cheeky i think you could do

    def differByOne(w1, w2):
        return len(w1) == len(w2) and sum(c1 != c2 for c1, c2 in zip(w1, w2)) == 1